sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/DerivedMetric.cs

```diff
@@ -16,7 +16,7 @@ namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.Filtering
     /// which defines which field to use as a value, and an aggregation which dictates the algorithm of arriving at
     /// a single reportable value within a second.
     /// </summary>
-    internal class DerivedMetric<TTelemetry>
+    internal class DerivedMetric<TTelemetry> where TTelemetry : DocumentIngress
```

Comment:
Why do you need to inherit if the existing code doesn't?

Reply:
The Filter class is using reflection to generate queries against the `DocumentIngress` classes (Request, Dependency, etc).
The concern is that if any of the DocumentIngress classes are trimmed, these generated queries may not work.
So I've added the compiler cues to the DocumentIngress class to prevent trimming.

The `where` constraint is going an extra mile. I want to codify that the Filter class is only used against the Document classes. The Filter class was only ever used with Documents, but that was previously not defined explicitly. 

As soon as the Filter class started using the `where` constraint, all the other related generic classes needed the same constraint.